AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/booking-details/booking-details.component.ts

```diff
@@ -126,6 +128,14 @@ export class BookingDetailsComponent implements OnInit, OnDestroy {
         }
     }
 
+    get canRetryConfirmation(): boolean {
+        if (!this.booking || this.booking.status !== BookingStatus.Failed) {
+            return false;
+        }
+        const scheduledTime = moment(this.booking.scheduled_date_time);
+        return scheduledTime.isAfter(moment(new Date()));
```

Comment:
What would happen if the page loads with a few seconds left and then extends over the time then the button was pressed. Should confirm hearing also call this?

Reply:
the button is bound to the property so if the time passes, like confirm, the button would disappear from the UI